### ISSUE TITLE
Build fails on undeclared proc_macro in signature

### ISSUE THREAD

**Issue by pepa65** [liljencrantz/crush#26]
rustc -V
`rustc 1.41.1 (f3e1a954d 2020-02-24)`

git clone https://github.com/liljencrantz/crush.git
cd crush; cargo build
```
  Downloaded sys-info v0.7.0
  Downloaded ssh2 v0.8.2
  Downloaded maplit v1.0.2
  Downloaded crossbeam v0.7.3
  Downloaded reqwest v0.10.1
  Downloaded termion v1.5.5
  Downloaded regex v1.3.1
  Downloaded serde_json v1.0.45
  Downloaded bytes v0.5.4
  Downloaded nix v0.17.0
  Downloaded rand v0.7.3
  Downloaded time v0.1.42
  Downloaded proc-macro2 v1.0.10
  Downloaded float-ord v0.2.0
  Downloaded syn v1.0.18
  Downloaded prost v0.6.1
  Downloaded toml v0.5.6
  Downloaded lalrpop-util v0.18.1
  Downloaded psutil v1.7.0
  Downloaded rustyline v5.0.3
  Downloaded users v0.9.1
  Downloaded prost-build v0.6.1
  Downloaded chrono v0.4.9
  Downloaded crossbeam-utils v0.7.0
  Downloaded url v2.1.1
  Downloaded nix v0.14.1
  Downloaded cc v1.0.45
  Downloaded aho-corasick v0.7.6
  Downloaded lalrpop v0.18.1
  Downloaded futures-util v0.3.2
  Downloaded crossbeam-queue v0.2.1
  Downloaded mime v0.3.16
  Downloaded dirs v2.0.2
  Downloaded regex-syntax v0.6.12
  Downloaded pin-project-lite v0.1.4
  Downloaded hyper-tls v0.4.1
  Downloaded getrandom v0.1.12
  Downloaded futures-core v0.3.2
  Downloaded ryu v1.0.2
  Downloaded native-tls v0.2.3
  Downloaded mime_guess v2.0.1
  Downloaded crossbeam-epoch v0.8.0
  Downloaded memchr v2.2.1
  Downloaded hyper v0.13.2
  Downloaded libssh2-sys v0.2.18
  Downloaded serde_urlencoded v0.6.1
  Downloaded parking_lot v0.10.2
  Downloaded http-body v0.3.1
  Downloaded tokio-tls v0.3.0
  Downloaded percent-encoding v2.1.0
  Downloaded unicode-segmentation v1.3.0
  Downloaded prost-derive v0.6.1
  Downloaded base64 v0.11.0
  Downloaded http v0.2.0
  Downloaded numtoa v0.1.0
  Downloaded rand_core v0.5.1
  Downloaded tokio v0.2.11
  Downloaded encoding_rs v0.8.22
  Downloaded unicode-width v0.1.6
  Downloaded utf8parse v0.1.1
  Downloaded bitflags v1.2.0
  Downloaded crossbeam-channel v0.4.0
  Downloaded rand_chacha v0.2.1
  Downloaded crossbeam-deque v0.7.2
  Downloaded itertools v0.8.2
  Downloaded fnv v1.0.6
  Downloaded tempfile v3.1.0
  Downloaded bit-set v0.5.1
  Downloaded term v0.5.2
  Downloaded pkg-config v0.3.17
  Downloaded unicase v2.6.0
  Downloaded futures-task v0.3.2
  Downloaded pin-project v0.4.8
  Downloaded openssl v0.10.28
  Downloaded autocfg v0.1.6
  Downloaded which v3.1.1
  Downloaded h2 v0.2.1
  Downloaded futures-io v0.3.2
  Downloaded parking_lot_core v0.7.2
  Downloaded openssl-sys v0.9.54
  Downloaded petgraph v0.5.0
  Downloaded diff v0.1.12
  Downloaded num-integer v0.1.41
  Downloaded matches v0.1.8
  Downloaded tower-service v0.3.0
  Downloaded httparse v1.3.4
  Downloaded pin-utils v0.1.0-alpha.4
  Downloaded anyhow v1.0.26
  Downloaded dtoa v0.4.5
  Downloaded want v0.3.0
  Downloaded futures-channel v0.3.2
  Downloaded num_cpus v1.12.0
  Downloaded num-traits v0.2.8
  Downloaded multimap v0.8.1
  Downloaded lock_api v0.3.4
  Downloaded openssl-probe v0.1.2
  Downloaded prost-types v0.6.1
  Downloaded libz-sys v1.0.25
  Downloaded ena v0.13.1
  Downloaded string_cache v0.7.5
  Downloaded ascii-canvas v2.0.0
  Downloaded dirs-sys v0.3.4
  Downloaded idna v0.2.0
  Downloaded version_check v0.9.1
  Downloaded bit-vec v0.5.1
  Downloaded tokio-util v0.2.0
  Downloaded fixedbitset v0.2.0
  Downloaded petgraph v0.4.13
  Downloaded c2-chacha v0.2.3
  Downloaded string_cache_codegen v0.4.4
  Downloaded precomputed-hash v0.1.1
  Downloaded foreign-types v0.3.2
  Downloaded string_cache_shared v0.3.0
  Downloaded futures-sink v0.3.2
  Downloaded either v1.5.3
  Downloaded indexmap v1.3.1
  Downloaded pin-project-internal v0.4.8
  Downloaded new_debug_unreachable v1.0.4
  Downloaded remove_dir_all v0.5.2
  Downloaded ppv-lite86 v0.2.6
  Downloaded phf_generator v0.7.24
  Downloaded foreign-types-shared v0.1.1
  Downloaded unicode-bidi v0.3.4
  Downloaded try-lock v0.2.2
  Downloaded phf_shared v0.7.24
  Downloaded unicode-normalization v0.1.12
  Downloaded siphasher v0.2.3
  Downloaded ordermap v0.3.5
  Downloaded fixedbitset v0.1.9
  Downloaded rand v0.6.5
  Downloaded rand_xorshift v0.1.1
  Downloaded rand_os v0.1.3
  Downloaded rand_hc v0.1.0
  Downloaded rand_jitter v0.1.4
  Downloaded rand_pcg v0.1.2
  Downloaded rand_isaac v0.1.1
  Downloaded rand_chacha v0.1.1
   Compiling libc v0.2.68
   Compiling cfg-if v0.1.10
   Compiling autocfg v0.1.6
   Compiling proc-macro2 v1.0.10
   Compiling unicode-xid v0.2.0
   Compiling lazy_static v1.4.0
   Compiling syn v1.0.18
   Compiling autocfg v1.0.0
   Compiling cc v1.0.45
   Compiling memchr v2.2.1
   Compiling log v0.4.8
   Compiling rand_core v0.4.2
   Compiling pkg-config v0.3.17
   Compiling bytes v0.5.4
   Compiling bitflags v1.2.0
   Compiling serde v1.0.104
   Compiling slab v0.4.2
   Compiling typenum v1.11.2
   Compiling fnv v1.0.6
   Compiling getrandom v0.1.12
   Compiling byteorder v1.3.2
   Compiling siphasher v0.2.3
   Compiling smallvec v1.2.0
   Compiling either v1.5.3
   Compiling anyhow v1.0.26
   Compiling semver-parser v0.7.0
   Compiling itoa v0.4.5
   Compiling futures-core v0.3.2
   Compiling version_check v0.9.1
   Compiling pin-project-lite v0.1.4
   Compiling string_cache_shared v0.3.0
   Compiling ppv-lite86 v0.2.6
   Compiling openssl v0.10.28
   Compiling regex-syntax v0.6.12
   Compiling foreign-types-shared v0.1.1
   Compiling byte-tools v0.3.1
   Compiling scopeguard v1.0.0
   Compiling futures-sink v0.3.2
   Compiling matches v0.1.8
   Compiling futures-io v0.3.2
   Compiling pin-utils v0.1.0-alpha.4
   Compiling httparse v1.3.4
   Compiling unicode-segmentation v1.3.0
   Compiling native-tls v0.2.3
   Compiling futures-task v0.3.2
   Compiling fixedbitset v0.1.9
   Compiling openssl-probe v0.1.2
   Compiling fake-simd v0.1.2
   Compiling ordermap v0.3.5
   Compiling remove_dir_all v0.5.2
   Compiling opaque-debug v0.2.3
   Compiling try-lock v0.2.2
   Compiling strsim v0.9.3
   Compiling fixedbitset v0.2.0
   Compiling new_debug_unreachable v1.0.4
   Compiling bit-vec v0.5.1
   Compiling precomputed-hash v0.1.1
   Compiling multimap v0.8.1
   Compiling diff v0.1.12
   Compiling ryu v1.0.2
   Compiling void v1.0.2
   Compiling percent-encoding v2.1.0
   Compiling nix v0.14.1
   Compiling encoding_rs v0.8.22
   Compiling tower-service v0.3.0
   Compiling mime v0.3.16
   Compiling dtoa v0.4.5
   Compiling nix v0.17.0
   Compiling numtoa v0.1.0
   Compiling unicode-width v0.1.6
   Compiling utf8parse v0.1.1
   Compiling base64 v0.11.0
   Compiling float-ord v0.2.0
   Compiling ordered_map v0.1.0 (/home/pp/git/crush/ordered_map)
   Compiling maplit v1.0.2
   Compiling thread_local v0.3.6
   Compiling rand_core v0.3.1
   Compiling rand_jitter v0.1.4
   Compiling rand_pcg v0.1.2
   Compiling rand_chacha v0.1.1
   Compiling rand v0.6.5
   Compiling crossbeam-utils v0.7.0
   Compiling num-traits v0.2.8
   Compiling crossbeam-epoch v0.8.0
   Compiling num-integer v0.1.41
   Compiling indexmap v1.3.1
   Compiling phf_shared v0.7.24
   Compiling itertools v0.8.2
   Compiling unicode-normalization v0.1.12
   Compiling http v0.2.0
   Compiling semver v0.9.0
   Compiling futures-channel v0.3.2
   Compiling foreign-types v0.3.2
   Compiling block-padding v0.1.5
   Compiling lock_api v0.3.4
   Compiling unicode-bidi v0.3.4
   Compiling unicase v2.6.0
   Compiling c2-chacha v0.2.3
   Compiling openssl-sys v0.9.54
   Compiling libz-sys v1.0.25
   Compiling sys-info v0.7.0
   Compiling libssh2-sys v0.2.18
   Compiling heck v0.3.1
   Compiling bit-set v0.5.1
   Compiling petgraph v0.4.13
   Compiling rand_xorshift v0.1.1
   Compiling rand_hc v0.1.0
   Compiling rand_isaac v0.1.1
   Compiling rustc_version v0.2.3
   Compiling idna v0.2.0
   Compiling want v0.3.0
   Compiling ena v0.13.1
   Compiling quote v1.0.3
   Compiling http-body v0.3.1
   Compiling aho-corasick v0.7.6
   Compiling futures-util v0.3.2
   Compiling generic-array v0.12.3
   Compiling rand_os v0.1.3
   Compiling iovec v0.1.4
   Compiling net2 v0.2.33
   Compiling num_cpus v1.12.0
   Compiling dirs v1.0.5
   Compiling which v3.1.1
   Compiling time v0.1.42
   Compiling atty v0.2.14
   Compiling parking_lot_core v0.7.2
   Compiling dirs-sys v0.3.4
   Compiling users v0.9.1
   Compiling termion v1.5.5
   Compiling psutil v1.7.0
   Compiling memoffset v0.5.3
   Compiling url v2.1.1
   Compiling crossbeam-queue v0.2.1
   Compiling crossbeam-channel v0.4.0
   Compiling mime_guess v2.0.1
   Compiling petgraph v0.5.0
   Compiling digest v0.8.1
   Compiling block-buffer v0.7.3
   Compiling regex v1.3.1
   Compiling rand_core v0.5.1
   Compiling mio v0.6.21
   Compiling term v0.5.2
   Compiling prost-build v0.6.1
   Compiling dirs v2.0.2
   Compiling parking_lot v0.10.2
   Compiling sha2 v0.8.1
   Compiling rand_chacha v0.2.1
   Compiling phf_generator v0.7.24
   Compiling lalrpop-util v0.18.1
   Compiling chrono v0.4.9
   Compiling tokio v0.2.11
   Compiling ascii-canvas v2.0.0
   Compiling rustyline v5.0.3
   Compiling rand v0.7.3
   Compiling string_cache_codegen v0.4.4
   Compiling ssh2 v0.8.2
   Compiling tempfile v3.1.0
   Compiling crossbeam-deque v0.7.2
   Compiling tokio-util v0.2.0
   Compiling string_cache v0.7.5
   Compiling tokio-tls v0.3.0
   Compiling crossbeam v0.7.3
   Compiling h2 v0.2.1
   Compiling serde_derive v1.0.104
   Compiling prost-derive v0.6.1
   Compiling pin-project-internal v0.4.8
   Compiling signature v0.1.0 (/home/pp/git/crush/signature)
error[E0433]: failed to resolve: use of undeclared type or module `proc_macro`
   --> signature/src/lib.rs:808:15
    |
808 |     metadata: proc_macro::TokenStream,
    |               ^^^^^^^^^^ use of undeclared type or module `proc_macro`

error[E0433]: failed to resolve: use of undeclared type or module `proc_macro`
   --> signature/src/lib.rs:809:12
    |
809 |     input: proc_macro::TokenStream,
    |            ^^^^^^^^^^ use of undeclared type or module `proc_macro`

error[E0433]: failed to resolve: use of undeclared type or module `proc_macro`
   --> signature/src/lib.rs:810:6
    |
810 | ) -> proc_macro::TokenStream {
    |      ^^^^^^^^^^ use of undeclared type or module `proc_macro`

error[E0433]: failed to resolve: use of undeclared type or module `proc_macro`
   --> signature/src/lib.rs:812:31
    |
812 |         Ok(res) | Err(res) => proc_macro::TokenStream::from(res),
    |                               ^^^^^^^^^^ use of undeclared type or module `proc_macro`

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0433`.
error: could not compile `signature`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

**Comment by liljencrantz** [comment]
This hapens because of https://blog.rust-lang.org/2020/03/12/Rust-1.42.html#use-proc_macrotokenstream-now-works

The simple fix is to upgrade to rustc 1.42. Is there a reason you don't want to? If there is a valid reason not to upgrade, I'm willing to keep the useless extra lines in there for a while, but if it's just inertia, I would prefer to keep the code simple and unmagical.

**Comment by pepa65** [comment]
No reason to upgrade, just ignorance... So I did `rustup upgrade` (ended up with 1.45) and now `cargo build` finishes no problem!
